src/signet.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <signet.h>
+
+#include <consensus/merkle.h>
+#include <consensus/params.h>
+#include <hash.h>
+#include <primitives/block.h>
+#include <script/interpreter.h>
+#include <script/standard.h>        // MANDATORY_SCRIPT_VERIFY_FLAGS
+#include <util/system.h>
+
+bool g_signet_blocks = false;
+CScript g_signet_blockscript;
```

Comment:
Likewise this can be in consensus params instead of being a new global.

Reply:
It can, but it would be wasteful; see detached comment.